packages/website/docs/plugins.md

```diff
@@ -3,21 +3,266 @@ id: plugins
+
+const appId = 'latency';
+const apiKey = '6be0576ff61c053d5f9a3225e2a90f76';
+const searchClient = algoliasearch(appId, apiKey);
+insightsClient('init', { appId, apiKey });
+
+const algoliaInsightsPlugin = createAlgoliaInsightsPlugin({ insightsClient });
+
+autocomplete({
+  // ...
+  plugins: [algoliaInsightsPlugin],
+});
+```
+:::note
+
+Plugins execute sequentially, in the defined order.
+
+:::
+
+### Building your own plugin
+
+An Autocomplete plugin is a simple object that implements the `AutocompletePlugin` interface.
+
+It can provide sources, react to state changes, and hook into various autocomplete lifecycle steps. It has access to setters, including the [Context API](context), allowing it to store and retrieve arbitrary data at any time.
+
+Let's create a plugin that adds a static list of GitHub repositories to the autocomplete results.
+
+```js
+const topGitHubRepositoriesPlugin = {
+  getSources() {
+    return [
+      {
+        getItems() {
+          return [
+            { name: 'algoliasearch-client-javascript', stargazersCount: 884 },
+            { name: 'algoliasearch-client-php', stargazersCount: 554 },
+            { name: 'hn-search', stargazersCount: 383 },
+          ];
+        },
+        getItemUrl({ item }) {
+          return `https://github.com/algolia/${item.name}`;
+        },
+        templates: {
+          header() {
+            return 'Discover our top 3 GitHub repositories';
+          },
+          item({ item }) {
+            return item.name;
+          },
+        },
+      },
+    ];
+  },
+};
+
+autocomplete({
+  // ...
+  plugins: [topGitHubRepositoriesPlugin],
+});
+```
+
+Now, if you're willing to package and distribute your plugin for other people to use, you might want to expose a function instead.
+
+```js
+function createTopGitHubRepositoriesPlugin({ repositories, username }) {
+  const sortedRepositories = repositories.sort(
+    (a, b) => b.stargazersCount - a.stargazersCount
+  );
+
+  return {
+    getSources() {
+      return sortedRepositories;
+    },
+    getItemUrl({ item }) {
+      return `https://github.com/${username}/${item.name}`;
+    },
+    templates: {
+      header() {
+        return `Discover our top ${repositories.length} GitHub repositories`;
+      },
+      item({ item }) {
+        return item.name;
+      },
+    },
+  };
+}
+
+const topGitHubRepositoriesPlugin = createTopGitHubRepositoriesPlugin({
+  username: 'algolia',
+  repositories: [
+    // ...
+  ],
+});
+```
+
+### Official plugins
+
+There are a few useful official plugins you can already use with Autocomplete.
+
+#### Recent searches
+
+The [`recent-searches`](createRecentSearchesPlugin) plugin lets you display a list of the latest searches the user made. It comes with a [pre-implemented version](createLocalStorageRecentSearchesPlugin) that connects with the user's [local storage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage).
+
+```js
```

Comment:
This is gonna have a maintenance cost to give full code snippets here. Don't you think that we should only introduce the 3 plugins but keep the code samples for their own API refs?

Reply:
Yeah that's what I said in my [PR description](https://github.com/algolia/autocomplete.js/pull/414#issue-561945229). The draft requested code snippets, but in fact that's scattering data around.

Reply:
Done in 871c1de.